### PR TITLE
RD-2038 Always select the first possible deployment in the table

### PR DESCRIPTION
**Pull request by Gelio**
Select the first possible deployment in the Deployments View widget if it is possible.

## System tests

https://jenkins.cloudify.co/blue/organizations/jenkins/Stage-UI-System-Test/detail/Stage-UI-System-Test/453/pipeline

Ran 2021-04-14 8:34 CEST

### CODE REVIEW

File: test/cypress/integration/widgets/deployments_view_spec.ts [hunk 1]
```diff
@@ -356,7 +356,7 @@ describe('Deployments View widget', () => {
 
             getDeploymentsViewTable().within(() => {
                 cy.log('Only top-level environments should be visible');
-                cy.contains('app-env').click();
+                cy.contains('app-env');
```
Comment (Gelio) [inline test/cypress/integration/widgets/deployments_view_spec.ts:359]: Clicking is no longer required, as it should be selected by default. The same applies to `db-env` below

**Review by kubama — APPROVED** [review]

File: widgets/common/src/deploymentsView/DeploymentsView.tsx [hunk 2]
```diff
@@ -45,15 +45,6 @@ export const DeploymentsView: FunctionComponent<DeploymentsViewProps> = ({
             }),
         {
             enabled: filteringByParentDeploymentResult.filterable,
-            onSuccess: data => {
```
Comment (Gelio) [inline widgets/common/src/deploymentsView/DeploymentsView.tsx:48]: Selecting a deployment only `onSuccess` caused no deployment when the request is refetching after going to a page that has some cached data (the previous cached deployments were shown, and the selection was wrong, and it was only updated after the new data arrived). Thus, I moved it to `render`

File: widgets/common/src/deploymentsView/DeploymentsView.tsx [hunk 3]
```diff
@@ -85,11 +76,19 @@ export const DeploymentsView: FunctionComponent<DeploymentsViewProps> = ({
         );
     }
 
-    const selectedDeployment = find(deploymentsResult.data.items, {
+    const context = toolbox.getContext();
+    const deployments = deploymentsResult.data.items;
+    const selectedDeployment = find(deployments, {
         // NOTE: type assertion since lodash has problems receiving string[] in the object
-        id: toolbox.getContext().getValue('deploymentId') as string | undefined
+        // eslint-disable-next-line react/destructuring-assignment
```
Comment (Gelio) [inline widgets/common/src/deploymentsView/DeploymentsView.tsx:83]: I don't have a clue why ESLint didn't like `context.getValue` below

**Review by qooban — APPROVED** [review]

**Comment by Gelio** [comment]
The system tests failed only due to coverage https://jenkins.cloudify.co/blue/organizations/jenkins/Stage-UI-System-Test/detail/Stage-UI-System-Test/453/pipeline/

I'll merge the PR

**Pull request merged**